website/content/labs/lab6.md

```diff
@@ -0,0 +1,176 @@
+
+By default, all files produced by a job are deleted once the job ends. However, we can tell the CI system to store one or more files for future use. These are called the artifacts of the job, and they can be downloaded after a job runs, or can be passed from one job to another.
+
+For example, we can tell GitLab to keep the `hello` binary after the `hello:build` job runs:
+
+    hello:build:
+        stage: build
+        script:
+          - make -C lab6/hello clean all
+        only: 
+          - master
+        artifacts:
+          expire_in: 1 hour
+          paths:
+            - lab6/hello/hello
+
+Notice how we specify that we want to artifact to expire in one hour, since we’re only going to use it in the subsequent `hello:test` job.
+
+To make sure the `hello:test` job receives that artifact, we need to define a dependency between the jobs:
+
+    hello:test:
+        stage: test
+        script:
+          - lab6/hello/hello
+        only: 
+          - master
+        dependencies:
+          - hello:build
+
+Now, do the following:
+
+* [5 points] Commit the updated `.gitlab-ci.yml` file, and copy-paste the URL of the resulting “passed” pipeline to the `tasks.txt` file. Notice how the `hello:test` task no longer builds the `hello` binary from scratch and, instead, downloads the artifact from the `hello:build` job. You should also be able to click on “Pipelines” -> “Jobs” and on a download icon next to the passed `hello:build` job to download the `hello` binary (you don’t need to do anything with it, just check you’re able to download it)
+* [20 points] Edit the `libgeometry:build` job so it will build both the libgeometry library and the tests. Next, edit the “libgeometry:tests” job so that it only runs the tests, using both the `libgeometry.so` file and the `tests/test-libgeometry` file from `libgeometry:build`. Copy-paste the URL of the resulting “passed” pipeline to the `tasks.txt` file.
+
+
+# Task 4: GitHub CI
+
+[10 points]
+
+Most version control systems include or integrate well with CI systems, and GitHub is no exception. It actually works with a number of different [CI systems](https://github.com/marketplace/category/continuous-integration), although a popular option is [Travis CI](https://travis-ci.org/)
+
+In this final task, you will take what you've learned about CI on GitLab, and see how it translates to a different context. So, this task is more about how to familiarize yourself with a new system (and knowing how to read documentation), than about CI specifics. In fact, all we want you to do in this task is the following:
+
+* Create a repository under your personal GitHub account called `cs220-lab6`.
+* Add the `hello/` files to the repository.
+* Setup Travis CI on your repository to build the `hello` binary and run the `hello` program.
+
+In the `tasks.txt` file, you must provide the following:
+
+* The URL of a successful build of your repository.
```

Comment:
The URL is actually useful here, but I think we should ask for both the URL and the output.